pallets/parachain-staking/src/lib.rs

```diff
@@ -2152,7 +2167,13 @@ pub mod pallet {
 		/// amount and updates `Unstaking` storage accordingly.
 		fn increase_lock(who: &T::AccountId, amount: BalanceOf<T>, more: BalanceOf<T>) -> Result<u32, DispatchError> {
 			ensure!(
-				pallet_balances::Pallet::<T>::free_balance(who) >= amount.into(),
+				pallet_balances::Pallet::<T>::reducible_balance(who, Preservation::Preserve, Fortitude::Polite)
```

Comment:
AFAIK it is not sufficient since the function is also used in the `join_delegator` extrinsic. The first check only validates if the user has enough balance for staking more, while the second check ensures that the user has enough balance to start a delegation.

Reply:
Oh ok I see how it's been used now. I think because of the new logic this function could also be simplified, but let's leave it like this to not increase the number of changes.